application/src/main/java/org/togetherjava/tjbot/features/chaptgpt/ChatGPTService.java

```diff
@@ -0,0 +1,69 @@
+package org.togetherjava.tjbot.features.chaptgpt;
+
+import com.theokanning.openai.OpenAiHttpException;
+import com.theokanning.openai.completion.chat.ChatCompletionRequest;
+import com.theokanning.openai.completion.chat.ChatMessage;
+import com.theokanning.openai.completion.chat.ChatMessageRole;
+import com.theokanning.openai.service.OpenAiService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Service used to communicate to OpenAI API to generate responses.
+ */
+public class ChatGPTService {
+    private static final Logger logger = LoggerFactory.getLogger(ChatGPTService.class);
+    private static final long TIMEOUT_DURATION = 10000L; // In milliseconds
```

Comment:
u can avoid the need for the comment and any confusion by making this a `Duration`:

```java
private static final Duration TIMEOUT = Duration.ofSeconds(10);
```